src/components/ClusterNodes/ColumnSelector.tsx

```diff
@@ -0,0 +1,129 @@
+                            <Setting className="icon-dim-16 setting-icon mr-5" />
+                            Columns
+                        </>
+                    )}
+                    {React.cloneElement(props.children[1])}
+                </>
+            ) : (
+                <>{props.children}</>
+            )}
+        </components.ValueContainer>
+    )
+}
+
+export default function ColumnSelector({
+    appliedColumns,
+    setAppliedColumns,
+}: {
+    appliedColumns: MultiValue<columnMetadataType>
+    setAppliedColumns: React.Dispatch<React.SetStateAction<MultiValue<columnMetadataType>>>
+}) {
+    const [selectedColumns, setSelectedColumns] = useState<MultiValue<columnMetadataType>>([])
+    const [isMenuOpen, setMenuOpen] = useState(false)
+    const [columnOptions, setColumnOptions] = useState<MultiValue<columnMetadataType>>([])
+
+    useEffect(() => {
+        setColumnOptions(columnMetadata.filter((columnData) => !columnData.isDisabled))
+        setSelectedColumns(appliedColumns)
+    }, [])
+
+    const handleApplySelectedColumns = (): void => {
+        setMenuOpen(false)
+        const _appliedColumns = [...selectedColumns].sort((a, b) => a['columnIndex'] - b['columnIndex'])
+        if (typeof Storage !== 'undefined') {
+            localStorage.appliedColumns = JSON.stringify(_appliedColumns)
+        }
+        setAppliedColumns(_appliedColumns)
+    }
+
+    const handleMenuState = (menuOpenState: boolean): void => {
+        if (menuOpenState) {
+            setSelectedColumns(appliedColumns)
+        }
+        setMenuOpen(menuOpenState)
+    }
+
+    const handleCloseFilter = (): void => {
+        handleMenuState(false)
+        setSelectedColumns(appliedColumns)
+    }
+
+    const MenuList = (props: any): JSX.Element => {
+        return (
+            <components.MenuList {...props}>
+                {props.children}
+                <div className="flex react-select__bottom bcn-0 p-8">
+                    <button className="flex cta apply-filter h-32 w-100" onClick={handleApplySelectedColumns}>
+                        Apply
+                    </button>
+                </div>
+            </components.MenuList>
+        )
+    }
+    return (
+        <ReactSelect
+            menuIsOpen={isMenuOpen}
+            name="columns"
+            value={selectedColumns}
+            options={columnOptions}
+            onChange={setSelectedColumns}
+            isMulti={true}
+            isSearchable={false}
+            closeMenuOnSelect={false}
+            hideSelectedOptions={false}
+            onMenuOpen={() => handleMenuState(true)}
+            onMenuClose={handleCloseFilter}
+            blurInputOnSelect={false}
+            autoFocus
+            components={{
+                Option: Option,
+                ValueContainer,
+                IndicatorSeparator: null,
+                ClearIndicator: null,
+                MenuList: (props) => <MenuList {...props} />,
```

Comment:
No need of `(props) => <MenuList {...props} />`  as it'll be same as passing MenuList directly.

Reply:
Done